x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go

```diff
@@ -72,97 +114,81 @@ func ValidatorExchangeRateCallback(k Keeper, ctx sdk.Context, args []byte, query
-	var lsmTokenDeposit recordstypes.LSMTokenDeposit
-	if inLSMLiquidStakeCallback {
-		var callbackData types.ValidatorExchangeRateQueryCallback
-		if err := proto.Unmarshal(query.CallbackData, &callbackData); err != nil {
-			return errorsmod.Wrapf(err, "unable to unmarshal validator exchange rate callback data")
-		}
-		lsmLiquidStake = *callbackData.LsmLiquidStake
-		lsmTokenDeposit = *lsmLiquidStake.Deposit
-	}
+	// Emit an event if the validator was slashed
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(hostZone.ChainId, ICQCallbackID_Validator,
+		"Previous Validator Exchange Rate: %v, Current Validator Exchange Rate: %v",
+		previousSharesToTokensRate, currentSharesToTokensRate))
 
-	// If the validator was not slashed, and we're in the callback of an LSM liquid stake,
-	// finish the transaction to mint the user their stTokens
-	if !validatorWasSlashed && inLSMLiquidStakeCallback {
-		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Validator,
-			"Validator was not slashed, finishing LSM liquid stake"))
-
-		// Finish the LSMLiquidStake with a temporary context so that the state changes can
-		// be discarded if it errors
-		err := utils.ApplyFuncIfNoError(ctx, func(ctx sdk.Context) error {
-			async := true
-			return k.FinishLSMLiquidStake(ctx, lsmLiquidStake, async)
-		})
-
-		// If finishing the transaction failed, emit an event and remove the LSMTokenDeposit
-		if err != nil {
-			errorMessage := fmt.Sprintf("lsm liquid stake callback failed after slash query: %s", err.Error())
-			EmitFailedLSMLiquidStakeEvent(ctx, hostZone, lsmTokenDeposit, errorMessage)
-			k.Logger(ctx).Error(errorMessage)
-
-			k.RecordsKeeper.RemoveLSMTokenDeposit(ctx, lsmLiquidStake.Deposit.ChainId, lsmLiquidStake.Deposit.Denom)
-		}
+	EmitValidatorExchangeRateChangeEvent(ctx, hostZone.ChainId, validator.Address, previousSharesToTokensRate, currentSharesToTokensRate)
 
-		return nil
-	}
+	return true, nil
+}
 
-	// If we're in the LSM liquid stake callback and there was a slash, reject the transaction by emitting an event
-	if inLSMLiquidStakeCallback {
-		// Emit an event to indicate that the transaction failed
-		errorMessage := fmt.Sprintf("validator was slashed - previous exchange rate: %s, current exchange rate: %s",
-			previousSharesToTokensRate, currentSharesToTokensRate)
+// Fails the LSM Liquid Stake if the query timed out
+func (k Keeper) LSMSlashQueryTimeout(ctx sdk.Context, hostZone types.HostZone, query icqtypes.Query) error {
+	var callbackData types.ValidatorExchangeRateQueryCallback
+	if err := proto.Unmarshal(query.CallbackData, &callbackData); err != nil {
+		return errorsmod.Wrapf(err, "unable to unmarshal validator exchange rate callback data")
+	}
+	lsmLiquidStake := *callbackData.LsmLiquidStake
 
-		EmitFailedLSMLiquidStakeEvent(ctx, hostZone, lsmTokenDeposit, errorMessage)
-		k.Logger(ctx).Error(errorMessage)
+	k.FailLSMLiquidStake(ctx, hostZone, lsmLiquidStake, "query timed out")
```

Comment:
Where are we checking if the query did actually time out?  This method should be called trigger timeout or something if the intention is it is a helper to force a failed query

Reply:
https://github.com/Stride-Labs/stride/pull/837/files/a2287f26558eb3f22a544ddb259168464a6077de#diff-10ab5e08a468a7d9d65c2b1bc816f16246abc31f96c5a56a056952fd70308424R44-R49